src/data/move.ts

```diff
@@ -581,11 +584,17 @@ export abstract class Move implements Localizable {
   /**
    * Sets the {@linkcode MoveFlags.G_MAX_MOVE} for the move
    * and {@linkcode moveTarget} to NEAR_ENEMY (g-max moves cannot target allies)
+   * and make the move not make contact
+   * and have the {@linkcode UseHigherAttackingStatAttr} to use the higher of the attacking stat
+   * and have the move use the {@linkcode GMaxPowerAttr}
    * @returns The {@linkcode Move} that called this function
    */
-  gMaxMove(): this {
+  gMaxMove(_signatureSpecies: Species): this {
     this.setFlag(MoveFlags.G_MAX_MOVE, true);
     this.moveTarget = MoveTarget.NEAR_ENEMY;
+    this.makesContact(false);
+    this.attr(UseHigherAttackingStatAttr);
+    this.attr(GMaxPowerAttr, _signatureSpecies);
```

Comment:
NIT: the argument shouldn't have the `_` prefix. This is associated with argument that aren't being used. It's used in line 597 though.

```suggestion
  gMaxMove(signatureSpecies: Species): this {
    this.setFlag(MoveFlags.G_MAX_MOVE, true);
    this.moveTarget = MoveTarget.NEAR_ENEMY;
    this.makesContact(false);
    this.attr(UseHigherAttackingStatAttr);
    this.attr(GMaxPowerAttr, signatureSpecies);
```